monitor/event_stats.hpp

```diff
@@ -0,0 +1,244 @@
+/*
+ * Copyright 2015+ Budnik Andrey <budnik27@gmail.com>
+ *
+ * This file is part of Elliptics.
+ *
+ * Elliptics is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Elliptics is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Elliptics.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef EVENT_STATS_HPP
+#define EVENT_STATS_HPP
+
+#include "cache/treap.hpp"
+
+#include <mutex>
+#include <algorithm>
+#include <functional> // not2
+#include <vector>
+#include <stack>
+
+namespace ioremap { namespace monitor {
+
+class mutex_lock_policy
+{
+	typedef std::mutex mutex_t;
```

Comment:
Please do not introduce such obscure types.
If you saw them in the code already, it would be better to change that